state-machine.md

```diff
@@ -51,6 +45,44 @@ type VMContext interface {
 
 If the execution completes successfully, changes to the state tree are saved. Otherwise, the message is marked as failed, and any state changes are reverted.
 
+```go
+func ApplyMessage(st StateTree, msg Message) MessageReceipt {
```

Comment:
This gets the gist of everything `ApplyMessage` does except tracking block gas limits and validation. It's possible that an implementation would implement these outside `ApplyMessage`, but it would be tricky. We track the gas consumed so far in the block and skip any message where `blockGasUsed + message.GasLimit > BlockGasLimit`. We validate:

1. Signature is valid
2. To != From
3. From is AccountActor
4. Sufficient Funds
5. Nonce isn't too low.
6. Nonce isn't too high.

Reply:
We shouldnt be requiring to != from, thats a perfectly valid thing to do.

explicitly checking that 'from' is an account actor feels redundant to checking the validity of the signature.